graylog2-web-interface/src/components/configurations/SearchesConfig.jsx

```diff
@@ -0,0 +1,288 @@
+    this.refs.searchesConfigModal.close();
+  },
+
+  render() {
+    const config = this.state.config;
+    const duration = moment.duration(config.query_time_range_limit);
+    const limit = this._isEnabled() ? `${config.query_time_range_limit} (${duration.format()})` : 'disabled';
+
+    let timerangeOptions = null;
+    let timerangeOptionsSummary = null;
+    if (this.state.timerangeOptionsList) {
+      timerangeOptions = this.state.timerangeOptionsList.map((option, idx) => {
+        const period = option.period;
+        const description = option.description;
+
+        return (
+          <div key={'timerange-option-' + idx}>
+            <Row>
+              <Col xs={4}>
+                <div className="input-group">
+                  <input type="text"
+                         className="form-control"
+                         value={period}
+                         onChange={this._onTimerangeOptionChange(idx, 'period')} />
+                  <span className="input-group-addon">
+                    {moment.duration(period).format()}
+                  </span>
+                </div>
+              </Col>
+              <Col xs={8}>
+                <div className="input-group">
+                  <input type="text"
+                         className="form-control"
+                         placeholder="Add description..."
+                         value={description}
+                         onChange={this._onTimerangeOptionChange(idx, 'description')} />
+                  <span className="input-group-addon">
+                    <i className="fa fa-trash" style={{cursor: 'pointer'}} onClick={this._onTimerangeOptionRemove(idx)} />
+                  </span>
+                </div>
+              </Col>
+            </Row>
+          </div>
+        );
+      });
+      timerangeOptionsSummary = this.state.timerangeOptionsList.map((option, idx) => {
+        const period = option.period;
+        const description = option.description;
+
+        return (
+          <span key={'timerange-options-summary-' + idx}>
+            <dt>{period}</dt>
+            <dd>{description}</dd>
+          </span>
+        );
+      });
+    }
+
+    return (
+      <div>
+        <h2>Search Configuration</h2>
+
+        <dl className={style.deflist}>
+          <dt>Query time range limit</dt>
+          <dd>{limit}</dd>
+          <dd>The maximum time users can query data in the past. This prevents users from accidentally creating queries which
+            span a lot of data and would need a long time and many resources to complete (if at all).</dd>
+        </dl>
+
+        <strong>Relative time range options</strong>
+
+        <dl className={style.deflist}>
+          {timerangeOptionsSummary}
+        </dl>
+
+        <IfPermitted permissions="clusterconfigentry:edit">
+          <Button bsStyle="info" bsSize="xs" onClick={this._openModal}>Update</Button>
+        </IfPermitted>
+
+        <BootstrapModalForm ref="searchesConfigModal"
```

Comment:
The component feels too big and I would actually move this form outside, just to make things more clear.